include/libtorrent/announce_entry.hpp

```diff
@@ -62,13 +91,7 @@ namespace libtorrent
 		// trackerid is sent).
 		std::string trackerid;
 
-		// if this tracker has returned an error or warning message
-		// that message is stored here
-		std::string message;
-
-		// if this tracker failed the last time it was contacted
-		// this error code specifies what error occurred
-		error_code last_error;
+		std::vector<announce_endpoint> endpoints;
```

Comment:
Do you think it would make sense (and simplify anything) to flatten this structure, and elevate each individual endpoint to the top level list of trackers? In some way endpoints can be treated as trackers in the same tier. But I suppose there's some special handling of endpoints with different endpoint families.